qiskit/primitives/base/base_sampler.py

```diff
@@ -200,3 +257,33 @@ def parameters(self) -> tuple[ParameterView, ...]:
             List of the parameters in each quantum circuit.
         """
         return tuple(self._parameters)
+
+
+BaseSampler = BaseSamplerV1
+
+
+class BaseSamplerV2:
+    """Sampler base class version 2.
+
+    A Sampler returns samples of quantum circuit outputs.
+
+    A Sampler implementation must treat the :meth:`.run` method ``shots=None`` kwarg
+    as using a default ``shots`` value.  The default value and methods to set it can
+    be determined by the Sampler implementor.
+    """
+
+    @abstractmethod
+    def run(self, pubs: Iterable[SamplerPubLike], shots: int | None = None) -> Job:
+        """Run and collect samples from each pub.
+
+        Args:
+            pubs: An iterable of pub-like objects. For example, a list of circuits
+                  or tuples ``(circuit, parameter_values)``.
+            shots: The total number of shots to sample for each :class:`.SamplerPub`.
```

Comment:
If SamplerV2 has an execution option and the number of shots of the execution option and that of `run` method differ, I guess that of `run` will be used. Am I correct?

E.g.,
```python
sampler = SamplerV2(options={"execution": {"shots": 10}})
result = sampler.run([circuit], shots=20).result()
# `result` contains 20 samples instead of 10 samples.
```